modules/Bio/EnsEMBL/Compara/PipeConfig/PairAligner_conf.pm

```diff
@@ -353,7 +353,7 @@ sub core_pipeline_analyses {
  	    },
  	    {  -logic_name => 'create_filter_duplicates_jobs', #factory
  	       -module     => 'Bio::EnsEMBL::Compara::RunnableDB::PairAligner::CreateFilterDuplicatesJobs',
- 	       -wait_for =>  [ 'update_max_alignment_length_before_FD' ],
+ 	       -wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', $self->o('pair_aligner_logic_name') ],
```

Comment:
What you propose could release the blocked jobs if a job in `LastZ` fails due to MEMLIMIT (it will be marked as done in `LastZ`). You need to either add `LastZ_himem` as well or, alternatively, make it wait for the factory, as no jobs in `check_no_partial_gabs` will be marked as DONE before all `LastZ` and `LastZ_himem` have completed anyway:
```suggestion
 	       -wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', 'create_pair_aligner_jobs' ],
```

Reply:
That didn't work. 
I ran the pipeline 3 times. The first time with:
`-wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', 'create_pair_aligner_jobs' ],`
the second time with:
`-wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', 'create_pair_aligner_jobs', $self->o('pair_aligner_logic_name') ],`
and the third with:
`-wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', $self->o('pair_aligner_logic_name') ],`
The last two worked. The first didn't.
I can revert back to:
`-wait_for =>  [ 'update_max_alignment_length_before_FD', 'check_no_partial_gabs', 'create_pair_aligner_jobs', $self->o('pair_aligner_logic_name') ],` if @carlacummins agrees, because this worked.

Reply:
The case you're outlining is almost impossible to occur since `create_pair_aligner_jobs` spends the majority of its runtime in `WRITE_OUTPUT`, producing new `LastZ` jobs. But if you really want to account for this extreme edge case, then sure - go with the final option outlined.

Reply:
In fact, the suggested change from @JAlvarezJarreta would actually be a step back IMO, since it's removing the `LastZ` blocker. The case where there are only completed jobs in `check_no_partial_gabs` and `create_pair_aligner_jobs` is a much more likely event and is one that I personally have seen already in e101

Reply:
So I wonder why the first solution doesn't work, and the second does, when, unless I'm missing something, they are exactly the same factory-fan-funnel scenario but we are applying two different `wait for` solutions (my proposal for the first one is exactly the same as the second one done).